internal/report/store.go

```diff
@@ -1,291 +1,45 @@
-		name = getClusterReportName(name)
-	}
-	getErr := s.client.Get(context.Background(), client.ObjectKey{Name: name}, &result)
-	if getErr != nil {
-		if errorMachinery.IsNotFound(getErr) {
-			return ClusterPolicyReport{}, constants.ErrResourceNotFound
-		}
-		return ClusterPolicyReport{}, getErr
-	}
-	return ClusterPolicyReport{
-		result,
-	}, nil
-}
+var SupportedTypes = [2]string{KUBERNETES, MEMORY}
 
-// Update namespaced PolicyReport
-func (s *PolicyReportStore) UpdatePolicyReport(report *PolicyReport) error {
-	err := s.client.Update(context.Background(), &report.PolicyReport)
-	if err != nil {
-		return err
-	}
-	summary, _ := report.GetSummaryJSON()
-	log.Info().
-		Dict("dict", zerolog.Dict().
-			Str("report name", report.GetName()).
-			Str("report ns", report.GetNamespace()).
-			Str("report resourceVersion", report.GetResourceVersion()).
-			Str("summary", summary),
-		).Msg("updated PolicyReport")
-	return nil
-}
+// PolicyReportStore caches the latest version of PolicyReports
+// There is 2 different stores available:
+//   - KubernetesPolicyReportStore, which uses K8s/etcd backend only
+//   - MemoryPolicyReportStore, which uses in-memory cache
+type PolicyReportStore interface {
+	// GetPolicyReport returns the Policy Report defined inside a given namespace.
+	// An empty PolicyReport is returned when nothing is found
+	GetPolicyReport(namespace string) (PolicyReport, error)
 
-// Update ClusterPolicyReport or PolicyReport. ns argument is used in case
-// of namespaced PolicyReport
-func (s *PolicyReportStore) UpdateClusterPolicyReport(report *ClusterPolicyReport) error {
-	err := s.client.Update(context.Background(), &report.ClusterPolicyReport)
-	if err != nil {
-		return err
-	}
-	summary, _ := report.GetSummaryJSON()
-	log.Info().
-		Dict("dict", zerolog.Dict().
-			Str("report name", report.GetName()).
-			Str("report ns", report.GetNamespace()).
-			Str("summary", summary),
-		).Msg("updated ClusterPolicyReport")
-	return nil
-}
+	// GetClusterPolicyReport gets the ClusterPolicyReport
+	GetClusterPolicyReport(name string) (ClusterPolicyReport, error)
 
-// Delete PolicyReport by namespace
-func (s *PolicyReportStore) RemovePolicyReport(namespace string) error {
-	if report, err := s.GetPolicyReport(namespace); err == nil {
-		err := s.client.Delete(context.Background(), &report.PolicyReport)
-		if err != nil {
-			return err
-		}
-	}
-	return nil
-}
+	// UpdatePolicyReport updates namespaced PolicyReport
+	UpdatePolicyReport(report *PolicyReport) error
 
-// Delete all namespaced PolicyReports
-func (s *PolicyReportStore) RemoveAllNamespacedPolicyReports() error {
-	err := s.client.DeleteAllOf(context.Background(), &polReport.PolicyReport{},
-		client.MatchingLabels(map[string]string{
-			LabelAppManagedBy: LabelApp,
-		}))
-	if err != nil {
-		return err
-	}
-	return nil
-}
+	// UpdateClusterPolicyReport updates a ClusterPolicyReport or PolicyReport.
+	// ns argument is used in case of namespaced PolicyReport
```

Comment:
there is no ns argument, nor this function updates a PolicyReport

Reply:
although I should've been more careful when moving the original godocs, this method will be removed with https://github.com/kubewarden/audit-scanner/pull/150#discussion_r1419164395